integr8ly/cloud-resource-operator
# Mgdapi 1635

## Overview

Jira: MGDAPI-1635

## Verification
In a perfect world, you would need to verify that CRO passes e2e tests on its own, CRO does well in case of the fresh install with RHOAM and doe not cause any issues in case of RHOAM upgrade. 

Sole CRO and E2Es:
- Get your hands on the BYOC cluster (to save nerves in case something goes wrong)
- Install CRO from this branch and run e2e tests 
- Verify all of them are passing 

> Below I'll be using "release" instead of "version of packagemanifests that refer to a new image" or how on earth it is named correctly? 

Fresh install of CRO with RHOAM
- get your hands on the BYOC cluster (since we would `useClusterStorage=false`) 
- use [the guide ](https://github.com/integr8ly/cloud-resource-operator/blob/master/Update_CRO_integreatly-operator.md#release-new-version-bundle-of-cloud-resource-operator) to release a new CRO version from this branch 
- use a master of RHOAM and make a new release to your own quay.io repo with a new release of CRO. There is also a [guide](https://github.com/RHCloudServices/integreatly-help/blob/master/guides/olm/install-upgrade-rhoam-bundle.md) but it seems every person has own preferred ways to generate bindles and index
- install a "new" RHOAM with a "new" CRO via olm 
- ensure installation and deletion succeeds 

Upgrade of CRO along with RHOAM upgrade 
- once again BYOC cluster 
- Here you would need to make bundles and index of CRO with `replaces` current version from integreatly. Or you can do the current release as well so it will not complain about nonexisting bundles 
- You would need to get two releases of RHOAM where latest one has a new CRO and replaces old one 
- Simulate upgrade scenario 
- Watch finalizers for CRO-related CRs get updated as well as installation succeeds 
- Ensure uninstallation completes with no issues 

## Checklist
